### ISSUE TITLE
`sub` displays #undef when `AbstractArray` subtype does not define getindex.

### ISSUE THREAD

**Issue by jakebolewski** [JuliaLang/julia#9997]
For example:
```julia
julia> t
5x5 Base.LinAlg.SymTridiagonal{Float64}:
 1.58856   0.13194    0.0        0.0        0.0
 0.13194   2.24674   -0.389461   0.0        0.0
 0.0      -0.389461   0.912113  -0.819463   0.0
 0.0       0.0       -0.819463   3.82956   -1.05546
 0.0       0.0        0.0       -1.05546    1.40946

julia> sub(t, [1:2])
2-element SubArray{Float64,1,Base.LinAlg.SymTridiagonal{Float64},(Array{Int64,1},),0}:
 #undef
 #undef

julia> sub(full(t), [1:2])
2-element SubArray{Float64,1,Array{Float64,2},(Array{Int64,1},),0}:
 1.58856
 0.13194
```

Shouldn't this throw a method error instead?

**Comment by timholy** [comment]
The `#undef` has nothing to do with `sub`, it's the display functionality. For example:
```julia
julia> t = SymTridiagonal(rand(5),rand(4))
5x5 Base.LinAlg.SymTridiagonal{Float64}:
 0.450664  0.594558  0.0       0.0        0.0     
 0.594558  0.863439  0.304512  0.0        0.0     
 0.0       0.304512  0.757572  0.520557   0.0     
 0.0       0.0       0.520557  0.0303424  0.155352
 0.0       0.0       0.0       0.155352   0.4823  

julia> t[3]
ERROR: indexing not defined for Base.LinAlg.SymTridiagonal{Float64}
 in getindex at abstractarray.jl:434

julia> a = sub(t, 1:2)
2-element SubArray{Float64,1,Base.LinAlg.SymTridiagonal{Float64},(UnitRange{Int64},),0}:
 #undef
 #undef

julia> a[1]
ERROR: indexing not defined for Base.LinAlg.SymTridiagonal{Float64}
 in anonymous at subarray2.jl:34
```
which is indeed the method error you were hoping for. IIRC there are some `try` statements in the display functionality.

All of these work if you first define
```julia
Base.getindex(A::SymTridiagonal, i::Real) = getindex(A, ind2sub(size(A), i)...)
```
before performing the above steps. It also doesn't happen with `sub(t, 1:2, 1)`.

But perhaps you're thinking that the display functionality needs to be changed? This isn't the first time this has proved confusing, see https://github.com/timholy/Grid.jl/pull/48 and https://github.com/tlycken/Interpolations.jl/issues/14.

**Comment by jakebolewski** [comment]
Sorry, the wording wasn't correct.  I think we should display something other than `#undef` in this case.

**Comment by mbauman** [comment]
This works now with #10525 since SymTridiagonal just got one-dimensional indexing for free.

**Comment by timholy** [comment]
I love the cascade of closed issues from #10525.
